chia/simulator/full_node_simulator.py

```diff
@@ -299,30 +317,34 @@ async def process_blocks(self, count: int, farm_to: bytes32 = bytes32([0] * 32))
         Returns:
             The total number of reward mojos for the processed blocks.
         """
-        rewards = 0
-        height = uint32(0)
+        if isinstance(timeout, _Default):
+            timeout = count * timeout_per_block
+            timeout += 1
 
-        if count == 0:
-            return rewards
+        with anyio.fail_after(delay=timeout):
+            rewards = 0
 
-        for _ in range(count):
-            block: FullBlock = await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
-            height = uint32(block.height)
-            rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
-
-        while True:
-            peak_height = self.full_node.blockchain.get_peak_height()
-            if peak_height is None:
-                raise RuntimeError("Peak height still None after processing at least one block")
+            if count == 0:
+                return rewards
 
-            if peak_height >= height:
-                break
+            for _ in range(count):
+                if guarantee_transaction_blocks:
+                    await self.farm_new_transaction_block(FarmNewBlockProtocol(farm_to))
+                else:
+                    await self.farm_new_block(FarmNewBlockProtocol(farm_to))
+                height = self.full_node.blockchain.get_peak_height()
+                assert height is not None
 
-            await asyncio.sleep(0.050)
+                rewards += calculate_pool_reward(height) + calculate_base_farmer_reward(height)
 
-        return rewards
+            return rewards
 
-    async def farm_blocks(self, count: int, wallet: Wallet) -> int:
+    async def farm_blocks_to_wallet(
```

Comment:
what is the diff between this and farm_blocks_to_puzzlehash except using wallet instead of specific puzzhash ?

Reply:
would it be better if we move the wait_for_coins_in_wallet to farm_blocks_to_puzzlehash and just avoid explicitly waiting for wallet in the tests ? or is there value in farm_blocks_to_puzzlehash without waiting for the wallet ?

(not saying this is relevant to this pr)

Reply:
If you are farming to a puzzle hash you may not even have a wallet.  A wallet could be made an optional parameter and if present we could verify the puzzle hash belongs to it and also wait for the wallet to sync etc.  I think mostly I wanted to shift away from fiddling with puzzle hashes.  In most of the tests I don't think we care about the puzzle hash.

Big picture, I think that the majority of times we farm we shouldn't even be doing that anyways.  We farm to a wallet (or non-zero puzzlehash) generally to get coins/funds into it so we can run the test.  something like `gimme_these_coins(wallet=wallet, coins=[1, 27, 13, 47])` would be a lot more representative in a test.  I imagine that could be implemented by sending from the prefarm coins and not "farming" at all.  Just processing blocks to complete transactions.